### PR TITLE
feat(fetch): implement abort

### LINKED ISSUE #5471: Add support for abortable fetch
Reference: https://developers.google.com/web/updates/2017/09/abortable-fetch

Also `Request` should have a `signal` property.

### PR DESCRIPTION
**Pull request by lucacasonato**
This commit introduces fetch aborting via an AbortSignal.

Closes #5471

### CODE REVIEW

File: tools/wpt/expectation.json [hunk 1]
```diff
@@ -1088,14 +1087,23 @@
...
-      ]
+      ],
+      "abort": {
+        "cache.https.any.html": false,
+        "general.any.html": [
+          "response.arrayBuffer() rejects if already aborted",
+          "response.blob() rejects if already aborted",
+          "response.formData() rejects if already aborted",
+          "response.json() rejects if already aborted",
+          "response.text() rejects if already aborted"
```
Comment (lucacasonato) [inline tools/wpt/expectation.json:1104]: These tests are not easy to solve without some significant refactors, so I will leave them failing for now. They are timing related (the promise resolves one microtask too late), and very unlikely to be noticed by a user.

Comment (bartlomieju) [reply tools/wpt/expectation.json:1103]: Is this related to `fetch` implementation or is it more general problem (like wrong timing of ops)?

**Review by bartlomieju — APPROVED** [review]
LGTM, cleanly implemented, no comments

File: tools/wpt/expectation.json (hunk 1 above)
Comment (lucacasonato) [reply tools/wpt/expectation.json:1103]: I am not quite sure yet

**Pull request merged**